Create index timeout on empty collection in ci testing

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```

[2021-06-03T16:04:36.028Z] =================================== FAILURES ===================================

[2021-06-03T16:04:36.028Z] _______ TestInsertBase.test_insert_after_create_index[get_simple_index1] _______

[2021-06-03T16:04:36.028Z] [gw2] linux -- Python 3.6.8 /usr/local/bin/python3

[2021-06-03T16:04:36.028Z] 

[2021-06-03T16:04:36.028Z] self = <test_insert.TestInsertBase object at 0x7f42ac852f28>

[2021-06-03T16:04:36.028Z] connect = <pymilvus.client.stub.Milvus object at 0x7f42e2485780>

[2021-06-03T16:04:36.028Z] collection = 'test_aBT1FzjX'

[2021-06-03T16:04:36.028Z] get_simple_index = {'index_type': 'IVF_FLAT', 'metric_type': 'L2', 'params': {'nlist': 128}}

[2021-06-03T16:04:36.028Z] 

[2021-06-03T16:04:36.028Z]     @pytest.mark.timeout(ADD_TIMEOUT)

[2021-06-03T16:04:36.028Z]     @pytest.mark.tags(CaseLabel.tags_smoke)

[2021-06-03T16:04:36.028Z]     def test_insert_after_create_index(self, connect, collection, get_simple_index):

[2021-06-03T16:04:36.028Z]         '''

[2021-06-03T16:04:36.028Z]         target: test build index insert after vector

[2021-06-03T16:04:36.028Z]         method: insert entities and build index

[2021-06-03T16:04:36.028Z]         expected: no error raised

[2021-06-03T16:04:36.028Z]         '''

[2021-06-03T16:04:36.028Z] >       connect.create_index(collection, field_name, get_simple_index)

[2021-06-03T16:04:36.028Z] 

[2021-06-03T16:04:36.028Z] entity/test_insert.py:146: 

[2021-06-03T16:04:36.028Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/stub.py:39: in inner

[2021-06-03T16:04:36.028Z]     return func(self, *args, **kwargs)

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/stub.py:728: in create_index

[2021-06-03T16:04:36.028Z]     return handler.create_index(collection_name, field_name, params, timeout, **kwargs)

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py:54: in handler

[2021-06-03T16:04:36.028Z]     return func(self, *args, **kwargs)

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py:805: in create_index

[2021-06-03T16:04:36.028Z]     status = future.result()

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/grpc/_channel.py:615: in result

[2021-06-03T16:04:36.028Z]     self._state.condition.wait, self._is_complete, timeout=timeout)

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/grpc/_common.py:140: in wait

[2021-06-03T16:04:36.028Z]     _wait_once(wait_fn, MAXIMUM_WAIT_TIMEOUT, spin_cb)

[2021-06-03T16:04:36.028Z] /usr/local/lib/python3.6/site-packages/grpc/_common.py:105: in _wait_once

[2021-06-03T16:04:36.028Z]     wait_fn(timeout=timeout)

[2021-06-03T16:04:36.028Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2021-06-03T16:04:36.028Z] 

[2021-06-03T16:04:36.028Z] self = <Condition(<unlocked _thread.RLock object owner=0 count=0 at 0x7f42dd2a25d0>, 0)>

[2021-06-03T16:04:36.028Z] timeout = 0.1

[2021-06-03T16:04:36.028Z] 

[2021-06-03T16:04:36.028Z]     def wait(self, timeout=None):

[2021-06-03T16:04:36.028Z]         """Wait until notified or until a timeout occurs.

[2021-06-03T16:04:36.028Z]     

[2021-06-03T16:04:36.028Z]         If the calling thread has not acquired the lock when this method is

[2021-06-03T16:04:36.029Z]         called, a RuntimeError is raised.

[2021-06-03T16:04:36.029Z]     

[2021-06-03T16:04:36.029Z]         This method releases the underlying lock, and then blocks until it is

[2021-06-03T16:04:36.029Z]         awakened by a notify() or notify_all() call for the same condition

[2021-06-03T16:04:36.029Z]         variable in another thread, or until the optional timeout occurs. Once

[2021-06-03T16:04:36.029Z]         awakened or timed out, it re-acquires the lock and returns.

[2021-06-03T16:04:36.029Z]     

[2021-06-03T16:04:36.029Z]         When the timeout argument is present and not None, it should be a

[2021-06-03T16:04:36.029Z]         floating point number specifying a timeout for the operation in seconds

[2021-06-03T16:04:36.029Z]         (or fractions thereof).

[2021-06-03T16:04:36.029Z]     

[2021-06-03T16:04:36.029Z]         When the underlying lock is an RLock, it is not released using its

[2021-06-03T16:04:36.029Z]         release() method, since this may not actually unlock the lock when it

[2021-06-03T16:04:36.029Z]         was acquired multiple times recursively. Instead, an internal interface

[2021-06-03T16:04:36.029Z]         of the RLock class is used, which really unlocks it even when it has

[2021-06-03T16:04:36.029Z]         been recursively acquired several times. Another internal interface is

[2021-06-03T16:04:36.029Z]         then used to restore the recursion level when the lock is reacquired.

[2021-06-03T16:04:36.029Z]     

[2021-06-03T16:04:36.029Z]         """

[2021-06-03T16:04:36.029Z]         if not self._is_owned():

[2021-06-03T16:04:36.029Z]             raise RuntimeError("cannot wait on un-acquired lock")

[2021-06-03T16:04:36.029Z]         waiter = _allocate_lock()

[2021-06-03T16:04:36.029Z]         waiter.acquire()

[2021-06-03T16:04:36.029Z]         self._waiters.append(waiter)

[2021-06-03T16:04:36.029Z]         saved_state = self._release_save()

[2021-06-03T16:04:36.029Z]         gotit = False

[2021-06-03T16:04:36.029Z]         try:    # restore state no matter what (e.g., KeyboardInterrupt)

[2021-06-03T16:04:36.029Z]             if timeout is None:

[2021-06-03T16:04:36.029Z]                 waiter.acquire()

[2021-06-03T16:04:36.029Z]                 gotit = True

[2021-06-03T16:04:36.029Z]             else:

[2021-06-03T16:04:36.029Z]                 if timeout > 0:

[2021-06-03T16:04:36.029Z] >                   gotit = waiter.acquire(True, timeout)

[2021-06-03T16:04:36.029Z] E                   Failed: Timeout >60.0s

[2021-06-03T16:04:36.029Z] 

[2021-06-03T16:04:36.029Z] /usr/local/lib/python3.6/threading.py:299: Failed

[2021-06-03T16:04:36.029Z] ------------------------------ Captured log setup ------------------------------

[2021-06-03T16:04:36.029Z] [2021-06-03 16:03:25-INFO-root]: {'index_type': 'IVF_FLAT', 'metric_type': 'L2', 'params': {'nlist': 128}} (test_insert.py:35)

[2021-06-03T16:04:36.029Z] ----------------------------- Captured stderr call -----------------------------

[2021-06-03T16:04:36.029Z] 

[2021-06-03T16:04:36.029Z] +++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++

[2021-06-03T16:04:36.029Z] 

[2021-06-03T16:04:36.029Z] ~~~~~~~~~~~~~~~~~~~~ Stack of Thread-4264 (139924347549440) ~~~~~~~~~~~~~~~~~~~~

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/threading.py", line 884, in _bootstrap

[2021-06-03T16:04:36.029Z]     self._bootstrap_inner()

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/threading.py", line 916, in _bootstrap_inner

[2021-06-03T16:04:36.029Z]     self.run()

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/threading.py", line 864, in run

[2021-06-03T16:04:36.029Z]     self._target(*self._args, **self._kwargs)

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/grpc/_channel.py", line 1126, in channel_spin

[2021-06-03T16:04:36.029Z]     event = state.channel.next_call_event()

[2021-06-03T16:04:36.029Z] 

[2021-06-03T16:04:36.029Z] ~~~~~~~~~~~~~~~~~~~~~ Stack of <unknown> (139926048257792) ~~~~~~~~~~~~~~~~~~~~~

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn

[2021-06-03T16:04:36.029Z]     reply.run()

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run

[2021-06-03T16:04:36.029Z]     self._result = func(*args, **kwargs)

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/execnet/gateway_base.py", line 967, in _thread_receiver

[2021-06-03T16:04:36.029Z]     msg = Message.from_io(io)

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/execnet/gateway_base.py", line 432, in from_io

[2021-06-03T16:04:36.029Z]     header = io.read(9)  # type 1, channel 4, payload 4

[2021-06-03T16:04:36.029Z]   File "/usr/local/lib/python3.6/site-packages/execnet/gateway_base.py", line 400, in read

[2021-06-03T16:04:36.029Z]     data = self._read(numbytes - len(buf))

[2021-06-03T16:04:36.029Z] 

[2021-06-03T16:04:36.029Z] +++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++

[2021-06-03T16:04:36.029Z] =========================== short test summary info ============================

[2021-06-03T16:04:36.029Z] FAILED entity/test_insert.py::TestInsertBase::test_insert_after_create_index[get_simple_index1]

[2021-06-03T16:04:36.029Z] !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!

[2021-06-03T16:04:36.029Z] !!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!
```
**Steps/Code to reproduce behavior**
https://internal.zilliz.com:18080/jenkins/blue/organizations/jenkins/milvus-ha-ci/detail/master/397/pipeline

**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Software conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)


**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/internal/proxynode/task.go
+++ b/internal/proxynode/task.go
@@ -3437,8 +3437,6 @@ func (gist *GetIndexStateTask) Execute(ctx context.Context) error {
 	getIndexStatesRequest := &indexpb.GetIndexStatesRequest{
 		IndexBuildIDs: make([]UniqueID, 0),
 	}
-	enableIndexBitMap := make([]bool, 0)
-	indexBuildIDs := make([]UniqueID, 0)
 
 	for _, segmentID := range allSegmentIDs {
 		describeSegmentRequest := &milvuspb.DescribeSegmentRequest{
@@ -3456,32 +3454,24 @@ func (gist *GetIndexStateTask) Execute(ctx context.Context) error {
 			return err
 		}
 		if segmentDesc.IndexID == matchIndexID {
-			indexBuildIDs = append(indexBuildIDs, segmentDesc.BuildID)
 			if segmentDesc.EnableIndex {
-				enableIndexBitMap = append(enableIndexBitMap, true)
-			} else {
-				enableIndexBitMap = append(enableIndexBitMap, false)
+				getIndexStatesRequest.IndexBuildIDs = append(getIndexStatesRequest.IndexBuildIDs, segmentDesc.BuildID)
 			}
 		}
 	}
 
-	log.Debug("proxynode", zap.Int("GetIndexState:: len of allSegmentIDs", len(allSegmentIDs)))
-	log.Debug("proxynode", zap.Int("GetIndexState:: len of IndexBuildIDs", len(indexBuildIDs)))
-	if len(allSegmentIDs) != len(indexBuildIDs) {
-		gist.result = &milvuspb.GetIndexStateResponse{
-			Status: &commonpb.Status{
-				ErrorCode: commonpb.ErrorCode_Success,
-				Reason:    "",
-			},
-			State: commonpb.IndexState_InProgress,
-		}
-		return err
+	gist.result = &milvuspb.GetIndexStateResponse{
+		Status: &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_Success,
+			Reason:    "",
+		},
+		State: commonpb.IndexState_Finished,
 	}
 
-	for idx, enableIndex := range enableIndexBitMap {
-		if enableIndex {
-			getIndexStatesRequest.IndexBuildIDs = append(getIndexStatesRequest.IndexBuildIDs, indexBuildIDs[idx])
-		}
+	log.Debug("ProxyNode GetIndexState", zap.Int("IndexBuildIDs", len(getIndexStatesRequest.IndexBuildIDs)), zap.Error(err))
+
+	if len(getIndexStatesRequest.IndexBuildIDs) == 0 {
+		return nil
 	}
 	states, err := gist.indexCoord.GetIndexStates(ctx, getIndexStatesRequest)
 	if err != nil {
@@ -3493,7 +3483,6 @@ func (gist *GetIndexStateTask) Execute(ctx context.Context) error {
 			Status: states.Status,
 			State:  commonpb.IndexState_Failed,
 		}
-
 		return nil
 	}
 
@@ -3503,17 +3492,8 @@ func (gist *GetIndexStateTask) Execute(ctx context.Context) error {
 				Status: states.Status,
 				State:  state.State,
 			}
-
 			return nil
 		}
-	}
-
-	gist.result = &milvuspb.GetIndexStateResponse{
-		Status: &commonpb.Status{
-			ErrorCode: commonpb.ErrorCode_Success,
-			Reason:    "",
-		},
-		State: commonpb.IndexState_Finished,
 	}
 
 	return nil

